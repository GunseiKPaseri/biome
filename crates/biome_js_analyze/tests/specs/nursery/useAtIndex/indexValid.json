[
  "array.at(-1)",
  "array[array.length - 0];",
  "array[array.length + 1]",
  "array[array.length + -1]",
  "foo[bar.length - 1]",
  "array?.[array.length - 1];",
  "array[array.length - 1] = 1",
  "array[array.length - 1] %= 1",
  "++ array[array.length - 1]",
  "array[array.length - 1] --",
  "delete array[array.length - 1]",
  "class Foo {bar; #bar; baz() {return this.#bar[this.bar.length - 1]}}",
  "([array[array.length - 1]] = [])",
  "({foo: array[array.length - 1] = 9} = {})"
]
