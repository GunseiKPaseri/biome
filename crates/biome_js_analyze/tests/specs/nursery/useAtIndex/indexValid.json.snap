---
source: crates/biome_js_analyze/tests/spec_tests.rs
assertion_line: 86
expression: valid.json
---
# Input
```cjs
array.at(-1)
```

# Input
```cjs
array[array.length - 0];
```

# Input
```cjs
array[array.length + 1]
```

# Input
```cjs
array[array.length + -1]
```

# Input
```cjs
foo[bar.length - 1]
```

# Input
```cjs
array?.[array.length - 1];
```

# Input
```cjs
array[array.length - 1] = 1
```

# Input
```cjs
array[array.length - 1] %= 1
```

# Input
```cjs
++ array[array.length - 1]
```

# Input
```cjs
array[array.length - 1] --
```

# Input
```cjs
delete array[array.length - 1]
```

# Input
```cjs
class Foo {bar; #bar; baz() {return this.#bar[this.bar.length - 1]}}
```

# Input
```cjs
([array[array.length - 1]] = [])
```

# Input
```cjs
({foo: array[array.length - 1] = 9} = {})
```
